sticker-utils/src/subcommands/pretrain.rs

```diff
@@ -387,33 +409,42 @@ impl StickerApp for PretrainApp {
             .value_of(SHUFFLE)
             .map(|v| v.parse().or_exit("Cannot parse shuffle buffer size.", 1));
         let parameters = matches.value_of(CONTINUE).map(ToOwned::to_owned);
-        let saver = matches
-            .value_of(SAVE_BATCH)
-            .map(|n| {
-                EpochAndBatchesSaver::new(
-                    "pretrain-",
-                    n.parse()
-                        .or_exit("Cannot parse number of batches after which to save", 1),
-                )
-                .into()
-            })
-            .unwrap_or_else(|| EpochSaver::new("pretrain-").into());
+        let eval_steps = matches
+            .value_of(EVAL_STEPS)
+            .unwrap()
+            .parse()
+            .or_exit("Cannot parse number of batches after which to save", 1);
         let logdir = matches.value_of(LOGDIR).map(ToOwned::to_owned);
 
+        // If steps is present, it overrides epochs.
+        let train_duration = if let Some(steps) = matches.value_of(STEPS) {
+            let steps = steps
+                .parse()
+                .or_exit("Cannot parse the number of training steps", 1);
+            TrainDuration::Steps(steps)
+        } else {
+            let epochs = matches
+                .value_of(EPOCHS)
+                .unwrap()
+                .parse()
+                .or_exit("Cannot parse number of training epochs", 1);
+            TrainDuration::Epochs(epochs)
+        };
```

Comment:
Maybe either print a message here clarifying what's the precedence if both are present or use [`Arg::conflicts_with`](https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.conflicts_with) above to avoid confusion here.

Reply:
`steps` uses `overrides_with`. IIRC can't use `conflicts_with` while also specifying a default value for `epochs` (because this option will then always be present and using `steps` will always conflict). Removing the default value for `epochs` makes the interface really weird (you always have to specify either `--epochs` or `--steps`.

Reply:
`Arg::default_value_if(STEPS,None,<defaullt val>)` may work

Reply:
This seems to work:

```rust
            .arg(
                Arg::with_name(EPOCHS)
                    .long("epochs")
                    .takes_value(true)
                    .value_name("N")
                    .help("Train for N epochs")
                    .default_value_if(STEPS, None,"2"),
            )
            .arg(
                Arg::with_name(STEPS)
                    .long("steps")
                    .value_name("N")
                    .help("Train for N steps")
                    .takes_value(true)
                    .conflicts_with(EPOCHS)
            )
```

Reply:
Thanks for sorting that out! I'll update the PR.

Reply:
No dice. For some reason `epochs` does not get a value when neither `steps` nor `epochs` is specified.

Reply:
Reading the docs, `.default_value_if(STEPS, None,"2"),` seems to mean: set the value of this option (epochs) to the given value if STEPS is specified.

Reply:
My bad, I only tested whether it works without providing config and data. I guess we can keep it as before and maybe add something like "overrides STEPS" to epoch's help string.

Reply:
> Reading the docs, `.default_value_if(STEPS, None,"2"),` seems to mean: set the value of this option (epochs) to the given value if STEPS is specified.

Weirdly enough, it seems to train if `STEPS` is specified which should, in turn, trigger the default value on `EPOCHS` which, in turn, should conflict with `STEPS`.